pkg/network/ebpf/c/tracer-stats.h

```diff
@@ -4,22 +4,25 @@
 #include "tracer.h"
 #include "tracer-maps.h"
 #include "tracer-telemetry.h"
+#include "cookie.h"
 
 static int read_conn_tuple(conn_tuple_t *t, struct sock *skp, u64 pid_tgid, metadata_mask_t type);
 
-static __always_inline u32 get_sk_cookie(struct sock *sk) {
+static __always_inline u32 get_stat_cookie(struct sock *sk) {
     u64 t = bpf_ktime_get_ns();
-    return (u32) ((u64)sk ^ t);
+    s64 cookie = get_socket_cookie(sk);
+    return (u32) (cookie ^ t);
 }
 
 static __always_inline conn_stats_ts_t* get_conn_stats(conn_tuple_t *t, struct sock *sk) {
     // initialize-if-no-exist the connection stat, and load it
     conn_stats_ts_t empty = {};
     __builtin_memset(&empty, 0, sizeof(conn_stats_ts_t));
-    empty.cookie = get_sk_cookie(sk);
+    empty.cookie = get_stat_cookie(sk);
```

Comment:
Why don't we use `get_socket_cookie` here?

Reply:
This solves a different problem where we can have a race between deletion (in the close path) and updating the connection stats for the same conn tuple, so this needs to be different.

Reply:
However, this would make it impossible to reconstruct the key for `conn_stats_t`, for protocol classification.

Reply:
> However, this would make it impossible to reconstruct the key for conn_stats_t, for protocol classification.

Reply:
> How does the race happen? Is conn_stats not deleted in the close path?

We don't delete the conn stats in the kretprobe for tcp_close so we could have a delete and then recreate happen.

Reply:
> I'm not sure what you mean by that.

I misunderstood the code. I thought the cookie was being used as a key into `conn_stats` map.